src/capabilities.ts

```diff
@@ -1,13 +1,18 @@
 import * as vscode from "vscode";
-import { getCapabilities } from "./requests";
+import { getCapabilities } from "./binary/requests";
 
-export const ON_BOARDING_CAPABILITY = "vscode.onboarding";
-export const NOTIFICATIONS_CAPABILITY = "vscode.user-notifications";
-export const VALIDATOR_CAPABILITY = "vscode.validator";
+export enum Capability {
+  ON_BOARDING_CAPABILITY = "vscode.onboarding",
+  VALIDATOR_CAPABILITY = "vscode.validator",
+}
+
+let isCapability: (c: Capability) => boolean;
+
+export function isCapabilityEnabled(c: Capability): boolean {
+  return isCapability ? isCapability(c) : false;
+}
 
-export function getCapabilitiesOnFocus(): Promise<{
-  isCapability: (string) => boolean;
-}> {
+export function fetchCapabilitiesOnFocus(): Promise<void> {
```

Comment:
why do you think this kind of API is better? it forces a temporal coupling between fetchCapabilitiesOnFocus and isCapabilityEnabled, you must call the first in order it to operate properly

Reply:
Yep, it goes for a global thing instead of propogating this function all the way down.